clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -335,7 +335,7 @@ class MarkDeviceFunction : public RecursiveASTVisitor<MarkDeviceFunction> {
       // instantiation as template functions. It means that
       // all functions used by kernel have already been parsed and have
       // definitions.
-      if (RecursiveSet.count(Callee)) {
+      if (RecursiveSet.count(Callee) && !Callee->isConstexpr()) {
```

Comment:
So I believe this change is still incorrect, since it permits calling a recursive function that isn't emitted at compile time.

Reply:
Take a look at the test I added - it shows that the recursive function **is** evaluated at compile time.

Reply:
I see/saw that test.  But also see the fact that the first test you submitted stopped being an error.  That needs to remain an error.